collector/pg_stat_walreceiver.go

```diff
@@ -0,0 +1,306 @@
+		[]string{"upstream_host", "slot_name"},
+		prometheus.Labels{},
+	)
+	statWalReceiverLastMsgSendTime = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, statWalReceiverSubsystem, "last_msg_send_time"),
+		"Send time of last message received from origin WAL sender",
+		[]string{"upstream_host", "slot_name"},
+		prometheus.Labels{},
+	)
+	statWalReceiverLastMsgReceiptTime = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, statWalReceiverSubsystem, "last_msg_receipt_time"),
+		"Send time of last message received from origin WAL sender",
+		[]string{"upstream_host", "slot_name"},
+		prometheus.Labels{},
+	)
+	statWalReceiverLatestEndLsn = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, statWalReceiverSubsystem, "latest_end_lsn"),
+		"Last write-ahead log location reported to origin WAL sender as integer",
+		[]string{"upstream_host", "slot_name"},
+		prometheus.Labels{},
+	)
+	statWalReceiverLatestEndTime = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, statWalReceiverSubsystem, "latest_end_time"),
+		"Time of last write-ahead log location reported to origin WAL sender",
+		[]string{"upstream_host", "slot_name"},
+		prometheus.Labels{},
+	)
+	statWalReceiverUpstreamNode = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, statWalReceiverSubsystem, "upstream_node"),
+		"Node ID of the upstream node",
+		[]string{"upstream_host", "slot_name"},
+		prometheus.Labels{},
+	)
+
+	pgStatWalColumnQuery = `
+	SELECT
+		column_name
+	FROM information_schema.columns
+	WHERE
+		table_name = 'pg_stat_wal_receiver' and
+		column_name = 'flushed_lsn'
+	`
+
+	pgStatWalReceiverQueryWithNoFlushedLSN = `
+	SELECT
+		trim(both '''' from substring(conninfo from 'host=([^ ]*)')) as upstream_host,
+		slot_name,
+		case status
+			when 'stopped' then 0
+			when 'starting' then 1
+			when 'streaming' then 2
+			when 'waiting' then 3
+			when 'restarting' then 4
+			when 'stopping' then 5 else -1
+		end as status,
+		(receive_start_lsn- '0/0') % (2^52)::bigint as receive_start_lsn,
+		receive_start_tli,
+		received_tli,
+		extract(epoch from last_msg_send_time) as last_msg_send_time,
+		extract(epoch from last_msg_receipt_time) as last_msg_receipt_time,
+		(latest_end_lsn - '0/0') % (2^52)::bigint as latest_end_lsn,
+		extract(epoch from latest_end_time) as latest_end_time,
+		substring(slot_name from 'repmgr_slot_([0-9]*)') as upstream_node
+	FROM pg_catalog.pg_stat_wal_receiver
+	`
+
+	pgStatWalReceiverQueryWithFlushedLSN = `
+	SELECT
+		trim(both '''' from substring(conninfo from 'host=([^ ]*)')) as upstream_host,
+		slot_name,
+		case status
+			when 'stopped' then 0
+			when 'starting' then 1
+			when 'streaming' then 2
+			when 'waiting' then 3
+			when 'restarting' then 4
+			when 'stopping' then 5 else -1
+		end as status,
+		(receive_start_lsn- '0/0') % (2^52)::bigint as receive_start_lsn,
+		receive_start_tli,
+		(flushed_lsn- '0/0') % (2^52)::bigint as flushed_lsn,
```

Comment:
If this is the only field that is different between the two queries, I would prefer to use something like fmt.Sprintf to stick this into a single query. There's a lot here and any future changes would need to be careful to make the same change in both queries.